### PR TITLE
 Dagger 2.11 compat generators for Application(Kotlin) and Activities(Kotlin&Java)

### PR DESCRIPTION
**Pull request by bhm**
Generator for Application component 
Generator for Kotlin and Java activities.

### CODE REVIEW

**Comment by MasterAccountIFM** [comment]
--- JIRA LINKS ---

**Comment by bhm** [comment]
Both of these PRs should be tested on a local machine.

Maybe I should provide a 2.11 sample too.

**Comment by PawelSwietochowski** [comment]
Tested on my Windows PC.
Since Application is making conflicts in Manifest mayby we shouldn't try to merge manifest.

File: android-studio-plugin/dagger_211/InfullAplicationKotlin_dagger211/root/src/mvp_folder/di/BuildersModule.kt.ftl [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package ${relativePackage}.di
+
+import dagger.Module;
```
Comment (PawelSwietochowski) [inline android-studio-plugin/dagger_211/InfullAplicationKotlin_dagger211/root/src/mvp_folder/di/BuildersModule.kt.ftl]: remove ;

File: android-studio-plugin/dagger_211/InfullAplicationKotlin_dagger211/root/src/mvp_folder/di/Module.kt.ftl [hunk 2]
```diff
@@ -0,0 +1,18 @@
+package ${relativePackage}.di;
+
+import android.content.Context;
```
Comment (PawelSwietochowski) [inline android-studio-plugin/dagger_211/InfullAplicationKotlin_dagger211/root/src/mvp_folder/di/Module.kt.ftl]: remove ;

File: android-studio-plugin/dagger_211/InfullAplicationKotlin_dagger211/root/src/mvp_folder/di/Module.kt.ftl [hunk 3]
```diff
@@ -0,0 +1,18 @@
+package ${relativePackage}.di;
```
Comment (PawelSwietochowski) [inline android-studio-plugin/dagger_211/InfullAplicationKotlin_dagger211/root/src/mvp_folder/di/Module.kt.ftl]: remove ;

File: android-studio-plugin/dagger_211/InfullAplicationKotlin_dagger211/root/src/mvp_folder/di/Module.kt.ftl [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+import android.content.Context;
+import ${relativePackage}.${applicationClass};
+
+import dagger.Module;
+import dagger.Provides;
```
Comment (PawelSwietochowski) [inline android-studio-plugin/dagger_211/InfullAplicationKotlin_dagger211/root/src/mvp_folder/di/Module.kt.ftl]: remove ;

File: android-studio-plugin/dagger_211/InfullAplicationKotlin_dagger211/root/src/mvp_folder/di/Module.kt.ftl [hunk 5]
```diff
@@ -0,0 +1,18 @@
+package ${relativePackage}.di;
+
+import android.content.Context;
+import ${relativePackage}.${applicationClass};
+
+import dagger.Module;
```
Comment (PawelSwietochowski) [inline android-studio-plugin/dagger_211/InfullAplicationKotlin_dagger211/root/src/mvp_folder/di/Module.kt.ftl]: remove ;

File: android-studio-plugin/dagger_211/InFullMvpActivityKotlin_dagger211/root/src/mvp_folder/di/Component.kt.ftl [hunk 6]
```diff
@@ -0,0 +1,16 @@
+package ${relativePackage}.di;
```
Comment (PawelSwietochowski) [inline android-studio-plugin/dagger_211/InFullMvpActivityKotlin_dagger211/root/src/mvp_folder/di/Component.kt.ftl]: remove ;

File: android-studio-plugin/dagger_211/InFullMvpActivityKotlin_dagger211/root/src/mvp_folder/di/Component.kt.ftl [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+
+import ${relativePackage}.${activityClass};
```
Comment (PawelSwietochowski) [inline android-studio-plugin/dagger_211/InFullMvpActivityKotlin_dagger211/root/src/mvp_folder/di/Component.kt.ftl]: remove ;

File: android-studio-plugin/dagger_211/InFullMvpActivityKotlin_dagger211/root/src/mvp_folder/di/Component.kt.ftl [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+import ${relativePackage}.di.${scopeClass}
+import dagger.Subcomponent
+import dagger.android.AndroidInjector
+
+import dagger.Component;
```
Comment (PawelSwietochowski) [inline android-studio-plugin/dagger_211/InFullMvpActivityKotlin_dagger211/root/src/mvp_folder/di/Component.kt.ftl]: remove ;

**Comment by bhm** [comment]
It looks like this is more work to resolve conflicted manifest, than to add application class it.

Removing it.
Clean up done.

**Comment by bhm** [comment]
I've simplified the Application template. We won't need to application module.

**Comment by rainqer** [comment]
we need templates to be in one folder, no further nesting. Nested folders templates do not work on osx.

**Comment by rainqer** [comment]
what's the difference between template dagger 2.11 and dagger 2.1x ?

**Pull request merged**